fendermint/app/src/main.rs

```diff
@@ -1,119 +1,51 @@
 // Copyright 2022-2024 Protocol Labs
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use anyhow::{anyhow, Context};
-use std::path::PathBuf;
-use std::str::FromStr;
-use std::sync::Arc;
-use trace4rs::config::{AppenderId, Policy, Target};
-use trace4rs::{
-    config::{self, Config, Format},
-    Handle,
-};
-
 pub use fendermint_app_options as options;
 pub use fendermint_app_settings as settings;
-use fendermint_app_settings::expand_tilde;
 
 mod cmd;
 
+fn init_tracing(opts: &options::Options) {
+    match &opts.log_dir {
+        Some(log_dir) => {
+            let file = tracing_appender::rolling::never(log_dir, "fendermint.log");
+
+            tracing_subscriber::fmt()
+                .json()
+                .compact()
+                .with_writer(file)
+                .with_target(false)
+                .with_file(true)
+                .with_line_number(true)
+                .with_max_level(opts.tracing_level())
+                .init();
+        }
+        None => {
+            tracing_subscriber::fmt()
```

Comment:
Can you add some comments on what the expectations are? If there is a log directory, is it not going to log anything to the console? What will it show in `docker logs` for example if we use the volumes? 

Can it be set up to log with `INFO` to `stdout` and with `DEBUG` or `TRACE` to the log files?

Reply:
Currently, when log directory is set, then traces all go to the logfile. What gets printed to STDOUT will be anything like println! or if a crate uses normal logger. 

> Can it be set up to log with INFO to stdout and with DEBUG or TRACE to the log files?
Something like that should be possible, what would be the usecase for that though? I would think that viewing the debug/trace logs we would want to show alongside the info+ logs as well. Or do you mean log INFO to both stdout and log files?

Reply:
Djeesus, I am having doubts of the `tracing` crate ecosystem, its not possible to have rotational logs with maximum filesize (but they have almost a year old [PR](https://github.com/tokio-rs/tracing/pull/2497) which hasn't been merged). The rotational logs are only based of time units (daily, hourly, etc) and it seems in order to use the `RollingFileAppender::builder` I can't use a non-blocking writer :(

Reply:
Ideally I'm looking for this kind of setup:
* in production we log to the console with an INFO level and to the files with a DEBUG levels, configurable individually: https://github.com/aakoshh/metronome/blob/develop/metronome/examples/resources/logback.xml
* in testing we log to the console: https://github.com/aakoshh/metronome/blob/develop/metronome/checkpointing/interpreter/specs/resources/logback-test.xml

Logging to the console has the benefit that it's easily available: you start the thing and don't have to dig around files to see _something_ happening, but you _can_ go to the files and the promtail Web UI for more details; you can also use `docker logs` to see some high level things.